### DISCUSSION TITLE
New component idea: collapsable chat widget

### DISCUSSION THREAD

**Discussion by benheckmann** [saadeghi/daisyui#1822] [Ideas / feature requests]
Hi!

In my current project, I need something similar to [this](https://github.com/Wolox/react-chat-widget) chat widget. I am now building it myself using daisyUI's text input, buttons and chat bubbles.

I think it would be neat if there was a complete component for that. With the current advancements in NLP, many people will want to build a chatbot into their project.

What do you think? I would be willing to make a contribution also.

**Comment by dosstx** [comment]
I would love to see this. I am surprised some of the other Tailwind sources do not have a chatbot component yet.

**Comment by saadeghi** [comment]
This is possible by using [dropdown](https://daisyui.com/components/dropdown/) and [chat bubble](https://daisyui.com/components/chat/) together.  
Here's an example:  
https://play.tailwindcss.com/mKJLY9wAG1  

It's as minimal as it can be. You can customize it as you want (colors, sizes, etc) and put it in a JS component
```html
<!-- dropdown -->
<div class="dropdown dropdown-end dropdown-top fixed bottom-10 right-10">
  <!-- button -->
  <label tabindex="0" class="btn-circle btn m-1">+</label>
  <!-- card -->
  <div tabindex="0" class="dropdown-content card w-80 bg-base-100 shadow-xl">
    <figure class="bg-primary p-6 text-2xl font-bold text-primary-content">Chat</figure>
    <div class="card-body p-4">
      <!-- chat bubble -->
      <div class="chat chat-start">
        <div class="chat-bubble">It's over Anakin, <br />I have the high ground.</div>
      </div>
      <!-- chat bubble -->
      <div class="chat chat-end">
        <div class="chat-bubble">You underestimate my power!</div>
      </div>
    </div>
    <div class="card-actions justify-end flex bg-base-200 p-4 flex-nowrap items-center">
      <!-- input -->
      <input type="text" placeholder="Type here" class="input input-bordered" />
      <!-- button -->
      <button class="btn">Send</button>
    </div>
  </div>
</div>
```

**Reply by benheckmann** [reply]
Very nice, thanks!
